uwblueprint/ccbc
# Derrick/general info form

## Notion ticket link

[Creator Profile: General Info form](https://www.notion.so/uwblueprintexecs/16fb020b93f641c8b384f76a1b3c769e?v=0f0df1f1b7074d2cb19b7eb7652d7d04&p=0bc58ded343540829ea575733f5da8bc&pm=c)



## Implementation description
* Fixed "previous" button logic on creator profile forms and added logic as necessary for navigation between forms using the buttons
* Updated `AddMultiSelect` component to handle errors when empty
* Added new component `AddProfilePictureField`, using the `StorageAPI` to return a firebase link and store that in the `CreatorProfileContext`, and also showing the link to the uploaded photo and size of the photo on the side. `localStorage` was used to persist the data and display when moving between pages with next and previous buttons since only the URL is stored in the context
* Added new component `RichTextEditor`, using `ReactQuill` component for the rich text editor. HTML of the text is stored in the context. Can also provide a `charLimit` property to limit the characters allowed
* Updated `CreatorInputField` component to allow for both non-required and required components
* Created `GeneralInfoForm` by using all the components above and correctly updated the `CreatorProfileContext` as needed



## Steps to test
1. Run it locally (`docker-compose up`)
2. Login and go to `http://localhost:3000/create-creator-profile`
3. Input whatever for the ContactInfoForm
4. Test permutations of inserting, deleting, and creating new items with the MultiSelect fields
5. Test the Rich Text Editor and character limit functionality
6. Test uploading your profile picture and seeing it update according to the Figma design
7.  Make sure the all fields stay after navigating to another page and back
8. Try different window sizes to see if formatting/styling is messed up
9. Test all permutations of the trying to go to the next page with required and non-required fields filled/not filled in.



## What should reviewers focus on?
* Lots of styling **within** the component props, not sure if this is a problem
* If window size changes, will there be issues with the hardcoded styling for components?
* Check permutations of multiselect input to test for the "required" field logic
* Can my new components I created be implemented to be more re-usable?
* Am I supposed to leave `yarn.lock` out of the PR
* is `localStorage` properly used for `AddProfilePicture` component? Not sure if next time they try to create a form, the information will still persist



## Checklist
